### PR TITLE
Add ability to view all who purchased an addon

### PR DESCRIPTION
**Pull request by NicoleZuckerman**
@CarlNelson style, plz.

@NorthIsUp take a look, make sure I'm not doing anything foolish?

Both: What should we be checking for when we surface the list of attendees?  We're doing that thing separately where we clear out all the addons that haven't been paid for...so getting the complete list HERE should be simple?

### CODE REVIEW

File: lindy/check_in/views.py [hunk 1]
```diff
@@ -409,3 +409,55 @@ def get_context_data(self, event_slug, ticket_id, user_id, **kwargs):
...
+    def get(self, *args, **kwargs):
+        redirect = get_redirect_to_login(self.request)
+        if redirect:
+            return redirect
+        if not self.request.user.is_focus_staff:
+            return HttpResponseNotFound('<h1>Page not found</h1>')
+        return super().get(self.request, *args, **kwargs)
+
+    def get_addon_attendees(self, addontype):
+        # NOTE: This function should do more work, like figure out who's paid up.
```
Comment (NicoleZuckerman) [inline lindy/check_in/views.py]: Adam: Can this be this simple?

File: lindy/check_in/templates/desk_check_in/addon_attendee_list.html [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+{% block content %}
+    {% if not user.is_authenticated or not user.is_focus_staff %}
+      you need to login
+    {% else %}
+        <div id="content-main" class="inner-two-columns">
+            <div class="row align-center align-middle">
+                {{ addontype.name }}
+                Total tickets sold:  {{addontype.count }}
+                {% if addontype.is_partnered %}
+                    Leads total: {{addontype.leaders_count }}
```
Comment (NicoleZuckerman) [inline lindy/check_in/templates/desk_check_in/addon_attendee_list.html]: Adam: Can I rely on addontype.leaders_count?

Comment (NorthIsUp) [inline lindy/check_in/templates/desk_check_in/addon_attendee_list.html]: Yes. This is what we use for limiting purchases

File: lindy/check_in/views.py (hunk 1 above)
Comment (NorthIsUp) [inline lindy/check_in/views.py]: [t for t in addon type.ticket_set.all() if t.is_paid]

Comment (NorthIsUp) [inline lindy/check_in/views.py]: *addon type should be addontype

Comment (NicoleZuckerman) [inline lindy/check_in/views.py]: But what if they've paid for THIS addon, but haven't paid for some OTHER addon that they just added to their cart? 
Besides, I thought we were filtering out the people who weren't paid in a manage command. So doesn't that make your addition unnecessary here, if we're subsequently turning off registration for this addon?

File: lindy/check_in/views.py [hunk 3]
```diff
@@ -409,3 +409,54 @@ def get_context_data(self, event_slug, ticket_id, user_id, **kwargs):
 
         )
         return kwargs
+
+
+class AddonsListView(TemplateView):
+    template_name = 'desk_check_in/addons_view_list.html'
+
+    def get(self, *args, **kwargs):
+        redirect = get_redirect_to_login(self.request)
```
Comment (NorthIsUp) [inline lindy/check_in/views.py]: this works but you can also use the `@login_required` decorator

Comment (NorthIsUp) [inline lindy/check_in/views.py]:
```
@method_decorator(login_required(login_url='/accounts/login/'), name='dispatch')
class AddonsListView(TemplateView):
```

**Pull request merged**